lib/ansible/plugins/inventory/netbox.py

```diff
@@ -262,7 +262,11 @@ def extract_device_role(self, host):
     def extract_config_context(self, host):
         try:
             if self.config_context:
-                url = self.api_endpoint + "/api/dcim/devices/" + str(host["id"])
+                # User 'device_role' vs. 'role' to differ between devices and virtual-machines
+                if 'device_role' in host:
+                    url = self.api_endpoint + "/api/dcim/devices/" + str(host["id"])
```

Comment:
problem with original, but still `str` is not portable, use `to_text`.

Reply:
Since this is a problem with the original script, it should be fixed for the whole script in a separate pull request.

Reply:
`str()` changed to `to_text()` for the whole script.